src/components/api/RoiContainer.tsx

```diff
@@ -16,6 +16,9 @@ export function RoiContainer({ target, children, id }: RoiComponentProps) {
       shortcut: ['delete', 'backspace'],
       handler: (event) => {
         if (event.isTrusted) {
+          roiDispatch({
+            type: 'CANCEL_ACTION',
+          });
```

Comment:
It's also better to have the "REMOVE_ACTION" action do it so that it is "atomic".

In general having successive sync dispatch calls should be avoided.